phpBB/phpbb/notification/type/mention.php

```diff
@@ -0,0 +1,407 @@
+<?php
+/**
+*
+* This file is part of the phpBB Forum Software package.
+*
+* @copyright (c) phpBB Limited <https://www.phpbb.com>
+* @license GNU General Public License, version 2 (GPL-2.0)
+*
+* For full copyright and license information, please see
+* the docs/CREDITS.txt file.
+*
+*/
+
+namespace phpbb\notification\type;
+
+/**
+* Post notifications class
+* This class handles notifications for replies to a topic
+*/
+class mention extends \phpbb\notification\type\post
+{
+    /**
+    * Get notification type name
+    *
+    * @return string
+    */
+    public function get_type()
+    {
+        return 'notification.type.mention';
+    }
+
+    /**
+    * Language key used to output the text
+    *
+    * @var string
+    */
+    protected $language_key = 'NOTIFICATION_MENTION';
+
+    /** @var \phpbb\user_loader */
+    protected $user_loader;
+
+    /** @var \phpbb\config\config */
+    protected $config;
+
+    /**
+    * Inherit notification read status from post.
+    *
+    * @var bool
+    */
+    protected $inherit_read_status = true;
+
+    static public $notification_option = false;
+
+    public function set_config(\phpbb\config\config $config)
+    {
+        $this->config = $config;
+    }
+
+    /**
+    * Initiate the class member with the object of class \phpbb]user_loader.
+    *
+    */
+    public function set_user_loader(\phpbb\user_loader $user_loader)
+    {
+        $this->user_loader = $user_loader;
+    }
+
+    /**
+    * Is available
+    */
+    public function is_available()
+    {
+        return $this->config['allow_topic_notify'];
+    }
+
+    /**
+    * Get the id of the item
+    *
+    * @param array $post The data from the post
+    * @return int The post id
+    */
+    static public function get_item_id($post)
+    {
+        return (int) $post['post_id'];
+    }
+
+    /**
+    * Get the id of the parent
+    *
+    * @param array $post The data from the post
+    * @return int The topic id
+    */
+    static public function get_item_parent_id($post)
+    {
+        return (int) $post['topic_id'];
+    }
+
+    /**
+    * Get the user's avatar
+    */
+    public function get_avatar()
+    {
+        return $this->user_loader->get_avatar($this->get_data('poster_id'), false, true);
+    }
+
+    /**
+    * Retrieves an ArrayIterator over the configuration values.
+    *
+    * @return \ArrayIterator An iterator over all config data
+    */
+    public function getIterator()
+    {
+        return new \ArrayIterator($this->config);
+    }
+
+    /**
+    * Extract user_ids of all users mentioned in the post.
+    *
+    * @return \Array An array of all mentioned userids.
+    */
+    public function find_users_for_notification($db, $users_to_notify = array())
```

Comment:
shouldn't be here. IIRC the notifications_table part is handled by the notification manager. Here the job is just to get a list of users that might be notified (look at the quote notification, it is extracting the users from the raw post: https://github.com/phpbb/phpbb/blob/master/phpBB/phpbb/notification/type/quote.php#L78)